cudapoa/src/application_parameters.cpp

```diff
@@ -0,0 +1,168 @@
+            break;
+        case 'f':
+            banded = false;
+            break;
+        case 'b':
+            band_width = std::stoi(optarg);
+            break;
+        case 'g':
+            graph_output_path = std::string(optarg);
+            break;
+        case 'n':
+            max_windows = std::stoi(optarg);
+            break;
+        case 'r':
+            gpu_mem_allocation = std::stod(optarg);
+            break;
+        case 'v':
+            print_version();
+        case 'h':
+            help(0);
+        default:
+            exit(1);
+        }
+    }
+
+    if (gpu_mem_allocation <= 0 || gpu_mem_allocation > 1.0)
+    {
+        std::cerr << "gpu-mem-alloc should be greater than 0 and less than 1.0" << std::endl;
+        exit(1);
+    }
+
+    if (consensus_mode < 0 || consensus_mode > 1)
+    {
+        std::cerr << "consensus_mode can only be 0 (consensus) or 1 (msa)" << std::endl;
+        exit(1);
+    }
+
+    verify_input_files(input_paths);
+}
+
+void ApplicationParameters::verify_input_files(std::vector<std::string>& input_paths)
+{
+    // Checks if the files are either all fasta or if one file is provided, it needs to be fasta or cudapoa
+    all_fasta = true;
+    for (auto& file_path : input_paths)
+    {
+        std::ifstream infile(file_path.c_str());
+        if (infile.good())
+        {
+            std::string firstLine;
+            std::getline(infile, firstLine);
+            if (firstLine.at(0) != '>')
+                all_fasta = false;
+            //break; TODO- Break here or not? It also provides input filepath verification @atadkase
+        }
+        else
+        {
+            std::cerr << "Invalid input file: " << file_path << std::endl;
+            exit(1);
+        }
+    }
+    if (input_paths.size() == 0 || (!all_fasta && input_paths.size() > 1))
+    {
+        std::cerr << "Invalid input. cudapoa needs input in either one cudapoa format file or in one/multiple fasta files." << std::endl;
+        help(1);
+    }
+}
+
+void ApplicationParameters::print_version(const bool exit_on_completion)
+{
+    std::cerr << claraparabricks_genomeworks_version() << std::endl;
+
+    if (exit_on_completion)
+    {
+        exit(1);
+    }
+}
+
+void ApplicationParameters::help(int32_t exit_code)
+{
+    std::cerr <<
+        R"(Usage: cudapoa [options ...]
+     options:
+        -i, --input
+            input in fasta/cudapoa format, can be used multiple times for multiple fasta files, but supports only one cudapoa file)"
+              << R"(
+        -m, --mode
+            consensus(0)/msa(1) [0])"
+              << R"(
+        -f, --full-alignment
+            uses full alignment if this flag is passed [banded alignment])"
+              << R"(
+        -g, --graph-output
+            output path for printing graph in DOT format [disabled])"
+              << R"(
+        -n, --max-windows
+            maximum number of windows to use from file)"
```

Comment:
not sure in documentation we use POA group or windows, they are closely related and within the code are used interchangeably, just wanted to ensure term window is defined in the docs.

Reply:
Changed to groups.